core/io/resource_loader.cpp

```diff
@@ -339,23 +339,31 @@ Ref<Resource> ResourceLoader::_load(const String &p_path, const String &p_origin
 		}
 	}
 #endif
-	ERR_FAIL_COND_V_MSG(found, Ref<Resource>(),
-			vformat("Failed loading resource: %s. Make sure resources have been imported by opening the project in the editor at least once.", p_path));
+
+	String error_message = vformat("Failed loading resource: %s. Make sure resources have been imported by opening the project in the editor at least once.", p_path);
+#ifdef TOOLS_ENABLED
+	if (Engine::get_singleton()->is_editor_hint()) {
+		// Remove part of the error message that is not relevant to the user (the project is already open in the editor).
+		error_message = vformat("Failed loading resource: %s.", p_path);
+	}
+#endif
+
+	ERR_FAIL_COND_V_MSG(found, Ref<Resource>(), error_message);
```

Comment:
Note that this will construct the `String` whether or not `found` is true, whereas the previous `ERR_FAIL` macro only constructs when true.

This is likely not a bottleneck, but might be better to enclose this whole section in:
```
if (found) {
...
}
```